### PR TITLE
New sub-trait to mock staking miner 

### PR DESCRIPTION
**Pull request by kianenigma**
This is zero logic change, just changing abstractions a bit. We extract part of the `EPM::Config` into a sub-trait called `MinerConfig`, so it can be mocked in other applications easier, without needing to mock the whole `EPM::Config`, which in itself implies the `system::Config` as well. 

polkadot companion: https://github.com/paritytech/polkadot/pull/5489

### CODE REVIEW

File: frame/election-provider-multi-phase/src/unsigned.rs [hunk 1]
```diff
@@ -255,45 +305,153 @@ impl<T: Config> Pallet<T> {
...
+	///
+	/// The weight is computed using `solution_weight`.
+	type MaxWeight: Get<Weight>;
+	/// Something that can compute the weight of a solution.
+	fn solution_weight(voters: u32, targets: u32, active_voters: u32, degree: u32) -> Weight;
+}
+
+/// A base miner, suitable to be used for both signed and unsigned submissions.
+pub struct Miner<T: MinerConfig>(sp_std::marker::PhantomData<T>);
+impl<T: MinerConfig> Miner<T> {
```
Comment (kianenigma) [inline frame/election-provider-multi-phase/src/unsigned.rs:423]: this entire impl block is old code that has been only moved, the git diff is not particularly good at showing that.

**Review by niklasad1 — APPROVED** [review]
LGTM

File: frame/election-provider-multi-phase/src/helpers.rs [hunk 2]
```diff
@@ -30,11 +33,21 @@ macro_rules! log {
 	};
 }
 
+#[macro_export]
+macro_rules! log_no_system {
```
[the remark this answers is not in the document]
Comment (kianenigma) [reply frame/election-provider-multi-phase/src/helpers.rs:38]: 0b892cf4dfb164bf3a4f34810fd73fbf638cb5e6

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**